`SDL_ELF_NOTE_DLOPEN` cannot be used with a toolchain not supporting variadic macros

`SDL_ELF_NOTE_DLOPEN` of `SDL_dlopennote.h` is currently designed to be used unconditionally.

https://github.com/libsdl-org/SDL/blob/45d65f6e1f51422c414e85f133849648918a8c09/include/SDL3/SDL_dlopennote.h#L102-L109

However, it cannot be used with an (older) toolchain not supporting variadic macros.

Compiling `testdlopennote.c` of [this PR](https://github.com/libsdl-org/SDL/pull/14212) fails with the error below (when using MSVC 6):
```
testdlopennote.c(13) : error C2059: syntax error : 'string'
```
Defining `SDL_DISABLE_DLOPEN_NOTES` does not help since it is already defined (Windows is not a unix platform).

--- a/include/SDL3/SDL_dlopennote.h
+++ b/include/SDL3/SDL_dlopennote.h
@@ -173,7 +173,7 @@
 #define SDL_DLNOTE_UNIQUE_NAME SDL_DLNOTE_JOIN(s_SDL_dlopen_note_, __LINE__)
 
 /**
- * Note that your application has dynamic shared library dependencies.
+ * Add a note that your application has dynamic shared library dependencies.
  *
  * You can do this by adding the following to the global scope:
  *
@@ -198,6 +198,8 @@
  * );
  * ```
  *
+ * This macro is not available for compilers that do not support variadic macro's.
+ *
  * \since This macro is available since SDL 3.4.0.
  *
  * \sa SDL_ELF_NOTE_DLOPEN_PRIORITY_SUGGESTED
@@ -219,7 +221,6 @@
 #elif defined(_MSC_VER) && _MSC_VER < 1400
 
 /* Variadic macros are not supported */
-#define SDL_ELF_NOTE_DLOPEN
 
 #else
 

